src/screens/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,160 @@
+import { useState } from "react";
+import { Container } from "@mui/material";
+import {
+  containerStyles,
+  CardStyle,
+  LogoBoxStyle,
+  IconStyle,
+  LogoTextStyle,
+  MainTextStyle,
+  AdditionalTextStyle,
+  LabelStyle,
+  InputStyle,
+  FormStyle,
+  InputProps,
+  ButtonStyle,
+} from "./Styles";
+import Card from "@mui/material/Card";
+import { Box } from "@mui/system";
+import Icon from "@mui/material/Icon";
+import Typography from "@mui/material/Typography";
+import { Button, TextField } from "@mui/material";
+import IconButton from "@mui/material/IconButton";
+import SVGinvisible from "../../assets/logos/isVisible";
+import MySvgFile from "../../assets/logos/authLogo";
+import { useForm, SubmitHandler } from "react-hook-form";
+
+const SignUp = () => {
+  const [isVisible, setIsVisible] = useState({
+    passwordInput: false,
+    confirmInput: false,
+  });
```

Comment:
Despite to created it in correct way. Good practice is not to handle complicated states (objects) and separate them as long until you realise that you really need them together. 

So please split them.